### PR TITLE
Ios 11 support

### LINKED ISSUE #939: Passcode/Touch ID required twice during open
### Steps to reproduce
1. Enable Passcode/Touch ID
2. Close and re-open app

### Expected behaviour
Touch ID once should log into app

### Actual behaviour
After Touch ID authenticates, another passcode screen comes up, forcing user to enter passcode

### Server configuration
**Web server:** nginx

**Database:** PostgreSQL

**PHP version:** 7.0

**ownCloud version:** 10

**Storage backend (external storage):** VPS SSD

### Client
**iOS version:** iOS 11 GM

**ownCloud app version:** Latest

**Device model:** iPhone 7+

#### Web server error log
```
Probably not relevant?
```

#### ownCloud log (data/owncloud.log)
```
Also probably not relevant...
```
 
### Tell us what could be improved:

I suspect this my have something race condition issues going on, not 100% sure though.

### PR DESCRIPTION
**Pull request by pablocarmu**
- [x] Compiled with Xcode9 without issues
- [x] Check changes in Swift4
- [X] Blanks over lists https://github.com/owncloud/ios/issues/905#issuecomment-313660230 [FIXED]
- [x] Background for instant uploads https://github.com/owncloud/ios/issues/905#issuecomment-313676014 [FIXED]
- [X] Fix #939 Double touch id https://github.com/owncloud/ios/issues/905#issuecomment-313678323 [FIXED]
- [X] Blanks bottom of the lists https://github.com/owncloud/ios/pull/959#issuecomment-355966812 [FIXED]
- [X] Touch id in iPhone X does not fit https://github.com/owncloud/ios/pull/959#issuecomment-355968153  [FIXED]
- [X] Landscape https://github.com/owncloud/ios/issues/905#issuecomment-352774104 [FIXED]
- [X] iPhone X splash screen https://github.com/owncloud/ios/pull/959#issuecomment-356370531 [FIXED]

### CODE REVIEW

File: Owncloud iOs Client/Tabs/FileTab/FilesViewController.m [hunk 1]
```diff
@@ -316,7 +321,7 @@ - (void)initFilesView {
          currentUser.userId == _mUser.userId)) {
         //We are changing of user
         //Show the file list in the correct place
-        [self.tableView setContentOffset:CGPointMake(0,0) animated:YES];
+//        [self.tableView setContentOffset:CGPointMake(0,0) animated:YES];
```
Comment (nasli) [inline Owncloud iOs Client/Tabs/FileTab/FilesViewController.m]: We can remove this comment

File: Owncloud iOs Client/Utils/SystemConstants.h [hunk 2]
```diff
@@ -18,9 +18,7 @@
 #define SCREEN_WIDTH ([[UIScreen mainScreen] bounds].size.width)
 #define SCREEN_HEIGHT ([[UIScreen mainScreen] bounds].size.height)
 #define SCREEN_MAX_LENGTH (MAX(SCREEN_WIDTH, SCREEN_HEIGHT))
-#define IS_IOS7 ([[[UIDevice currentDevice] systemVersion] floatValue] >= 7 && [[[UIDevice currentDevice] systemVersion] floatValue] < 8)
-#define IS_IOS8 ([[[UIDevice currentDevice] systemVersion] floatValue] >= 8 && [[[UIDevice currentDevice] systemVersion] floatValue] < 9)
-#define IS_IOS9 ([[[UIDevice currentDevice] systemVersion] floatValue] >= 9)
+#define IS_IOS9or10 ([[[UIDevice currentDevice] systemVersion] floatValue] == 9 || [[[UIDevice currentDevice] systemVersion] floatValue] == 10)
```
Comment (nasli) [inline Owncloud iOs Client/Utils/SystemConstants.h]: We could remove this and use the condition (!IS_IOS11) instead

**Comment by jesmrec** [comment]
Blanks over list are fixed, but now the blanks are in the bottom of the list

Tested with iPhone X, v11.2.1

With iPhone 6SPlus, v11.0.3, the file list is correctly displayed

**Comment by jesmrec** [comment]
Related with the touch id, the label should be changed in case of iPhoneX, so it does not support touch id, but face id. Should be a renaming of the label enough in this case?

**Comment by jesmrec** [comment]
Bug https://github.com/owncloud/ios/issues/905#issuecomment-313678323  keeps happening

**Comment by jesmrec** [comment]
iPhoneX face id, double touch id and landscape are fixed

**Comment by jesmrec** [comment]
The problem related with the blanks is fixed in iOS11, but appears in iPhone v10

**Comment by jesmrec** [comment]
In iPhoneX, The splash screen does not fit the screen

**Comment by jesmrec** [comment]
All bugs fixed

**Comment by nasli** [comment]
Released within 3.7.2

**Pull request merged**